evm/p2p/kademlia.py

```diff
@@ -516,7 +516,7 @@ def ping(self, node: Node) -> bytes:
         async def _find_node(node_id, remote):
             self.wire.send_find_node(remote, node_id)
             candidates = await self.wait_neighbours(remote)
-            if len(candidates) == 0:
+            if not candidates:
```

Comment:
I'm not sure I agree with pylint here; the original check makes it clear that we expect candidates to be of type sequence, so we'll get an exception if by accident we get a different type whereas the new check may fail silently if we get something unexpected here.

Reply:
If we're keeping it for type safety, wouldn't it then be even better to do
```python
if candidates == []:
```
To nail down the type completely and avoid all accidents? with len() == 0 it'll work with dicts, strings, etc.

The argument for not checking type at all is that this code is already covered with static type checks, so changing return type will fail mypy tests, and it is redundant to check the type several times.

Reply:
I seem to remember working on projects that preferred explicit length checks instead of relying on the fact that empty sequences are false, and I also personally prefer that, but it looks like PEP-8 recommends otherwise[1], so it's probably better if we stick to it

[1] For sequences, (strings, lists, tuples), use the fact that empty sequences are false.